terraform/cross-device/project.tf

```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
```

Comment:
This is not needed because you already pass the output of a `google_project` datasource as `var.project_id`. Remove and refactor accordingly.

Reply:
It's used for getting the project number during the pubsub creation (https://github.com/GoogleCloudPlatform/federated-learning/pull/336/files#diff-0a96bed47710a16197ba8216e9be8cb0a0d6047ebf552419a1da6aa22bfbe0c4R30)